source/channel_bootstrap.c

```diff
@@ -1377,10 +1397,10 @@ int aws_server_bootstrap_destroy_socket_listener(struct aws_server_bootstrap *bo
         AWS_CONTAINER_OF(listener, struct server_connection_args, listener);
 
     AWS_LOGF_DEBUG(AWS_LS_IO_CHANNEL_BOOTSTRAP, "id=%p: releasing bootstrap reference", (void *)bootstrap);
+
     aws_socket_stop_accept(listener);
     aws_socket_clean_up(listener);
-    aws_mem_release(bootstrap->allocator, server_connection_args);
-    s_server_bootstrap_release(bootstrap);
+    s_server_connection_args_release(server_connection_args);
```

Comment:
Again, not sure if this could cause a bug, but seems like a good idea to move `aws_socket_clean_up(listener);` into the release() function?